src/main.rs

```diff
@@ -41,10 +41,21 @@ fn main() -> CliResult {
 fn run_floki_from_args(args: &Cli) -> Result<(), Error> {
     debug!("Got command line arguments: {:?}", &args);
 
-    let config = FlokiConfig::from_file(&args.config_file)?;
-    debug!("Got configuration {:?}", &config);
+    let environ = environment::Environment::gather()?;
+    debug!("Got environment {:?}", &environ);
+
+    let (floki_root, config_file) = match &args.config_file {
+        Some(config_file) => (environ.current_directory.to_path_buf(), config_file.clone()),
+        None => {
+            let config_file = find_floki_yaml(&environ.current_directory)?;
+            // config_file is guaranteed to be a file, not a directory
+            (config_file.parent().unwrap().to_path_buf(), config_file)
```

Comment:
May as well use `?` instead of `unwrap` here.

Reply:
As above - I've moved to use `expect`.